File: src/openfl/events/IEventDispatcher.nim
import ./Event

type
  IEventDispatcher* = concept x
    x.addEventListener(string, proc (event: Event), bool, int, bool)
    x.dispatchEvent(Event) is bool
    x.hasEventListener(string) is bool
    x.removeEventListener(string, proc (event: Event), bool)
    x.willTrigger(string) is bool

File: kfl/src/main/kotlin/openfl/events/IEventDispatcher.kt
package openfl.events

interface IEventDispatcher {
    fun addEventListener(
        type: String,
        listener: (Event) -> Unit,
        useCapture: Boolean = false,
        priority: Int = 0,
        useWeakReference: Boolean = false
    )

    fun dispatchEvent(event: Event): Boolean
    fun hasEventListener(type: String): Boolean
    fun removeEventListener(
        type: String,
        listener: (Event) -> Unit,
        useCapture: Boolean = false
    )
    fun willTrigger(type: String): Boolean
}
